addons/OPT/Beam/fn_openDialog.sqf

```diff
@@ -0,0 +1,96 @@
+#include "macros.hpp"
+#include "\opt_core\dialogs\OPTDef.hpp"
+
+
+private _success = createDialog "opt_beamDialog";
+
+private _display = findDisplay IDD_BEAM;
+private _ctrlList = _display displayCtrl IDC_BEAM_LIST;
+private _ctrlMap = _display displayCtrl IDC_BEAM_MAP;
+
+
+private _markerSelected = createMarkerLocal ["beamSelected", [0,0,0]];
+_markerSelected setMarkerAlphaLocal 0;
+_markerSelected setMarkerTypeLocal "selector_selectedMission";
+_markerSelected setMarkerColorLocal "ColorBlue";
+_markerSelected setMarkerSizeLocal [1.2, 1.2];
+
+
+private _mapMarkers = [];
+{
+	private _name = _x getVariable "name";
+	private _unit = _x getVariable "unit";
+	private _id = _x getVariable "id";
+	private _idx = _ctrlList lbAdd _name;
+	_ctrlList lbSetData [_idx, _id]; // Set the generated ID as invisible data for more robust reference later
+	private _marker = createMarkerLocal [ _id, getPos _unit];
+	_marker setMarkerTextLocal _name;
+	_marker setMarkerTypeLocal "selector_selectable";
+	_marker setMarkerColor "ColorBlue";
+	_mapMarkers pushBack _marker;
+} foreach (call OFUNC(getPoints));
+
+_display setVariable ["markers", _mapMarkers];
+_display setVariable ["markerSelected", _markerSelected];
+
+
+
+// EH params are passed as array in _this ([<units>,<pos>,<alt>,<shift>]) and in special variables _units, _pos, _alt, _shift
+["beamSelect", "onMapSingleClick", {
+	private _display = _this select 4;
+	private _markers = _display getVariable "markers";
+	private _markerSelected = _display getVariable "markerSelected";
+	private _list = _this select 5;
+	private _map = _this select 6;
+	_map ctrlRemoveAllEventHandlers "Draw";
+	// Check if clicked close enough:
+	private _screenCoordClick = _map ctrlMapWorldToScreen _pos;
+	private _closestMarker = nil;
+	private _closestDistance = 999;
+	// Instead of using world distance ( = meters), we use screenspace. This way, the proximity calculation does not depend on the mapScale / zoom level.
+	{
+		_screenCoordMarker = _map ctrlMapWorldToScreen getMarkerPos _x;
+		private _distance = _screenCoordMarker distance _screenCoordClick;
+		// Distance is generously the area a default marker occupies in screenspace, use as hard cutoff
+		if (_distance < 0.03 && _distance < _closestDistance) then {
+			_closestMarker = _x;
+			_closestDistance = _distance;
+		};
+	} forEach _markers;
+
+	if (isNil "_closestMarker") exitWith {
+		_markerSelected setMarkerAlphaLocal 0;
+		_list lbSetCurSel -1;
+	};
+
+	// hint format ["Clicked on %1 at %2", _closestMarker, _closestDistance];
+	[_closestMarker, "map"] call OFUNC(selectBeam);
```

Comment:
Wie löst sich `OFUNC()` eigentlich auf? einfach in `opt_fnc_selectBeam`? Oder ist das Modul selbst auch noch verbaut ( `opt_beam_fnc_selectBeam` )?

Reply:
Nee das Modul ist nicht mit drin. Das ist genau der Punkt. Das Modul ist nur bei FUNC mit dabei.
Dh es das erste ist richtig

Reply:
Also wie rufe ich Fnc aus anderen Modulen auf? Bzw. wie verhindere ich, dass Funktionsnamen durch versch. Module doppelt belegt werden, wenn alle einfach `opt_fnc_name` sind?

Reply:
Wenn du Funktionen aus anderen Modulen aufrufen willst, musst du die entsprechende Funktion als API Funktion deklarieren und dann via `OFUNC` aufrufen.

Da muss man dann aber immer selber aufpassen, dass man keine vorhandene API Funktion überschreibt.